cfme/tests/test_login.py

```diff
@@ -89,3 +89,27 @@ def test_update_password(context, request, appliance):
         appliance.server.login(user)
 
     user.delete()
+
+
+@pytest.mark.parametrize('context', [ViaUI])
+def test_leading_whitespace_password(context, request, appliance):
+    """ Test password with leading whitespace """
+
+    # First, create a temporary new user
+    username = 'user_temp_{}'.format(fauxfactory.gen_alphanumeric(4).lower())
+    password = " password"
+    new_creds = Credential(principal=username, secret=password)
+    user_group = appliance.collections.groups.instantiate(description="EvmGroup-vm_user")
+    user = appliance.collections.users.create(
+        name=username,
+        credential=new_creds,
+        groups=user_group
+    )
+
+    # Try to login with new user
+    logged_in_page = appliance.server.login(user)
+    assert logged_in_page.is_displayed
+    logged_in_page.logout()
+
+    # Delete the user
+    user.delete()
```

Comment:
REQUIRED: if assert or something else  happen before this line. User won't be removed. Please handle this by either creating appropriate yield fixture or adding request.addfinalizer.